superset/utils/core.py

```diff
@@ -749,6 +752,34 @@ def __exit__(  # pylint: disable=redefined-outer-name,unused-variable,redefined-
             logger.exception(ex)
 
 
+class TimerTimeout:
+    def __init__(self, seconds: int = 1, error_message: str = "Timeout") -> None:
+        self.seconds = seconds
+        self.error_message = error_message
+        self.timer = threading.Timer(seconds, _thread.interrupt_main)
+
+    def __enter__(self) -> None:
+        self.timer.start()
+
+    def __exit__(  # pylint: disable=redefined-outer-name,unused-variable,redefined-builtin
+        self, type: Any, value: Any, traceback: TracebackType
+    ) -> None:
+        self.timer.cancel()
+        if type is KeyboardInterrupt:  # raised by _thread.interrupt_main
```

Comment:
Surprised that this is a `KeyboardInterrupt`

Reply:
Me too!